app/api_contract.py

```diff
@@ -0,0 +1,385 @@
+import datetime
+import enum
+import uuid
+from typing import Annotated
+
+from fastapi import FastAPI, Header, status
+from pydantic import BaseModel, EmailStr, Field, JsonValue
+
+app = FastAPI()
+
+
+JWTToken = Annotated[
+    str,
+    Header(
+        alias="Authroization",
+        title="JWT Token",
+        description="The JWT token used for authorization in `Bearer <token>` format",
+    ),
+]
+
+CloudUserToken = Annotated[
+    str,
+    Header(
+        alias="Authroization",
+        title="FinOps for Cloud user token",
+        description="The token used for authorization in `Bearer <token>` format",
+    ),
+]
+
+
+class ErrorResponse(BaseModel):
+    detail: str
+
+
+class CommonPrivateFields:
+    id: uuid.UUID
+    created_at: datetime.datetime
+    updated_at: datetime.datetime
+    deleted_at: Annotated[
+        datetime.datetime | None,
+        Field(
+            default=None,
+            examples=[None],
+            description="If not null, the timestamp the object was deleted at",
+        ),
+    ]
+
+
+class CreateOrgRequest(BaseModel):
+    display_name: Annotated[str, Field(examples=["Apple Inc."])]
+
+
+class OrganizationDetail(BaseModel, CommonPrivateFields):
+    display_name: Annotated[str, Field(examples=["Apple Inc."])]
+    external_id: Annotated[
+        str,
+        Field(
+            description="the ID of the account of the MPT platform which owns this organization"
+        ),
+    ]
+    cloudspend_id: uuid.UUID
+
+
+class CreateUserRequest(BaseModel):
+    email: EmailStr
+    organization_id: uuid.UUID
+    display_name: Annotated[str, Field(examples=["Tim Cook"])]
+
+
+class UserDetail(BaseModel, CommonPrivateFields):
+    email: EmailStr
+    organization_id: uuid.UUID
+    display_name: Annotated[str, Field(examples=["Tim Cook"])]
+    cloudspend_id: uuid.UUID
+
+
+class CloudType(enum.Enum):
+    AWS_ROOT = "aws_root"
+    AWS_LINKED = "aws_linked"
+    AZURE_TENANT = "azure_tenant"
+    AZURE_SUBSCRIPTION = "azure_subscription"
+    GCP = "gcp"
+    ALIBABA = "alibaba"
+    DATABRICKS = "databricks"
+    KUBERNETES = "kubernetes"
```

Comment:
I know we don't want to enable these for now (this is documented bellow), included them for completeness